### PR TITLE
.NET 5.0

### PR DESCRIPTION
**Pull request by bruno-garcia**
- [x] Upgrade dependencies (other than Sentry which is done)
- [x] Migrations needed with new EF?
- [x] Use new docker images and deploy

Can be browsed at: https://stg.nugt.net

### CODE REVIEW

File: src/NuGetTrends.Api/NuGetTrends.Api.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk.Web">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net5.0</TargetFramework>
```
Comment (joaopgrassi) [inline src/NuGetTrends.Api/NuGetTrends.Api.csproj:4]: Nice 😎

File: src/NuGetTrends.Api/Startup.cs [hunk 2]
```diff
@@ -51,7 +51,9 @@ public void ConfigureServices(IServiceCollection services)
             services
                 .AddDbContext<NuGetTrendsContext>(options =>
                 {
-                    options.UseNpgsql(Configuration.GetConnectionString("NuGetTrends"));
+                    options
+                        .UseNpgsql(Configuration.GetConnectionString("NuGetTrends"))
+                        .UseSnakeCaseNamingConvention();
```
Comment (joaopgrassi) [inline src/NuGetTrends.Api/Startup.cs]: Hum not sure we want to switch to this. The last time we tried, there was several issues related to it dropping primary keys and changing identity columns... not sure it's a good idea 😅

Also: This convention is changing the EF Migration history table columns. Doing migrations operations later is failing with: 

```
Failed executing DbCommand (9ms) [Parameters=[], CommandType='Text', CommandTimeout='30']
SELECT migration_id, product_version
FROM "__EFMigrationsHistory"
ORDER BY migration_id;
Npgsql.PostgresException (0x80004005): 42703: column "migration_id" does not exist
```
There's a bug open on that repo: https://github.com/efcore/EFCore.NamingConventions/issues/1. The "workaround" is to manually update the columns in the db first. But I don't like that much tbh. 

Also: I noticed that there's a migration named `EFUpgrade` I think that is from the last attempt? I think we didn't run that one right? It looks quite destructive. :(

**Review by joaopgrassi — COMMENTED** [review]
I opened a PR (#121) on this branch as we talked to fix the EF caos with left-over migrations from the last upgrade (3.1). Now there's no pending migration (adding a new just creates a empty file). Also, the exception is gone and the API at least runs normally.

**Comment by bruno-garcia** [comment]
In the evening I'll run it in staging and we see how it goes for a while before merging/running live
Thanks!

**Review by joaopgrassi — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Comment by bruno-garcia** [comment]
It's running fine, lets merge.
In a few days I'll make a release

**Pull request merged**